homeassistant/components/websocket_api.py

```diff
@@ -477,17 +477,20 @@ def handle_get_config(self, msg):
         self.to_write.put_nowait(result_message(
             msg['id'], self.hass.config.as_dict()))
 
-    def handle_get_panels(self, msg):
+    def handle_get_panels(self, msg, request):
```

Comment:
Wouldn't it be better to store if we want ES6 in the constructor of the connection? That way we don't have to add another parameter to each callback function.

Reply:
In the constructor we are not aware of the es5/es6/auto config setting.

Reply:
Sorry, meant the `handle` method.

Reply:
First, in order for websocket to decided on es6-vs-es5, `_is_latest` would have to move to websocket_api while depending on data filled by frontend. I think this is a bad dependency.

Regardless, the `es6` variable would have to be passed to handlers which is just as bad as passing the request.

Reply:
We can store it in the instance as `self.frontend_is_latest = True` but it's fine for now, we can tweak it later.